src/app/core/templates/activities/activity-create/activity-create.component.ts

```diff
@@ -103,29 +107,37 @@ export class ActivityCreateComponent implements OnInit {
         }
     }
 
-    // function for using datepicker in form for creating activities
-    openDatePicker() {
-        this.datepicker.open = true;
-    }
-
     wrongInput(ErrorMessage) {
         this.loading = false;
         return alert(ErrorMessage);
     }
 
+    onFileSelect(event) {
+        if (event.target.files.length > 0) {
+            const file = event.target.files[0];
+            this.uploadForm.get('image').setValue(file);
+        } else {
+            this.uploadForm.get('image').setValue('No Image');
+        }
+    }
+
+    groupFilterCallback(group, query) {
+        return group.canOrganize === true;
+    }
+
     // function called when a new activity is submitted
     submit() {
         this.loading = true;
 
         let hasCoverImage = false;
-        if (this.uploadme !== "No Image") {
+        if (this.uploadForm.get('image').value !== 'No Image') {
             hasCoverImage = true;
         }
 
         const fd = new FormData();
         if (hasCoverImage) {
-            // @ts-ignore
-            const file = $('#activityCreate-cover')[0].files[0];
+            const file = this.uploadForm.get('image').value;
+
             if (!file.type.startsWith('image/')) {
                 return this.wrongInput('Non-image formats are not supported as pictures for activities!');
```

Comment:
Should we also incorporate a check here for the file type, or is that redundant?